### PR TITLE
Warn if port is not specified in connectTo

### PR DESCRIPTION
**Pull request by jeffluo35**
cjdroute segfaults if there a port isn't specified, this makes it so that it warns the user.

### CODE REVIEW

**Comment by lgierth** [comment]
That's a good one -- thanks a lot. We can make it even better by giving the user more information to fix the problem, e.g. "connection [foo] must be $IP:$PORT, or [$IP]:$PORT for IPv6.".

**Comment by jeffluo35** [comment]
Should that be a separate log message, or part of the same one?

**Comment by lgierth** [comment]
Can just replace the existing message

File: client/Configurator.c [hunk 1]
```diff
@@ -216,12 +216,19 @@ static void udpInterface(Dict* config, struct Context* ctx)
...
                 char* lastColon = CString_strrchr(key->bytes, ':');
 
-                if (!Sockaddr_parse(key->bytes, NULL)) {
-                    // it's a sockaddr, fall through
-                } else if (lastColon) {
-                    // try it as a hostname.
-                    Log_critical(ctx->logger, "Couldn't add connection [%s], "
-                                                "hostnames aren't supported.", key->bytes);
+                if (lastColon) {
+                    if (!Sockaddr_parse(key->bytes, NULL)) {
```
Comment (kpcyrd) [inline client/Configurator.c]: I'm not sure this is correct, this assumes `Sockaddr` has a colon in it, I don't know if that's the case. Assuming it's the [sockaddr struct][1] there's a good chance it doesn't contain a colon but is still valid. I would appreciate if somebody more familiar with that part of the source could confirm this :)

[1]: http://www.retran.com/beej/sockaddr_inman.html

Comment (kpcyrd) [inline client/Configurator.c]: I digged into the code and you're right, it's actually expecting a string here, sorry.

**Pull request merged**